utils/s3/client.cc

```diff
@@ -385,10 +403,110 @@ class client::upload_sink final : public client::upload_sink_base {
     }
 };
 
+future<> client::upload_sink_base::upload_part(std::unique_ptr<upload_sink> piece_ptr) {
```

Comment:
Well, in that case the `upload_part(memory_data_sink_buffers)` would need to get moved to `upload_sink` too. I wanted to avoid that not to add public exporters for private members just to make these two calls work :(